Simplified and leverage the Microsoft Identity App Sync .NET tool (#112799)
Azure/azure-content: articles/active-directory/develop/tutorial-blazor-server.md

- The tenant-id or domain of the Azure Active Directory associated with your Azure Account

## Create the app using the .NET CLI

```dotnetcli
mkdir <new-project-folder>
cd <new-project-folder>
dotnet new blazorserver --auth SingleOrg --calls-graph
```

## Install the Microsoft Identity App Sync .NET Tool

```dotnetcli
dotnet tool install --global msidentity-app-sync
```

This tool will automate the following tasks for you:

- Register your application in Azure Active Directory
  - Create a secret for your registered application
  - Register redirect URIs based on your launchsettings.json
- Initialize the use of user secrets in your project
- Store your application secret in user secrets storage
- Update your appsettings.json with the client-id, tenant-id, and others.

.NET Tools extend the capabilities of the dotnet CLI command. To learn more about .NET Tools, see [.NET Tools](/dotnet/core/tools/global-tools).

For more information on user secrets storage, see [safe storage of app secrets during development](/aspnet/core/security/app-secrets).

## Use the Microsoft Identity App Sync Tool

> Run the following command to register your app in your tenant and update the .NET configuration of your application. Provide the username/upn belonging to your Azure Account (for instance, username@domain.com) and the tenant ID or domain name of the Azure Active Directory associated with your Azure Account. If you use an account that is signed in in either Visual Studio, Azure CLI, or Azure PowerShell, you'll benefit from Single Sign On (SSO).

Reviewer: ```suggestion
Run the following command to register your app in your tenant and update the .NET configuration of your application. Provide the username/upn belonging to your Azure Account (for instance, `username@domain.com`) and the tenant ID or domain name of the Azure Active Directory associated with your Azure Account. If you use an account that is signed in in either Visual Studio, Azure CLI, or Azure PowerShell, you'll benefit from single sign-on (SSO).
```

  - [Application administrator](../roles/permissions-reference.md#application-administrator)
  - [Application developer](../roles/permissions-reference.md#application-developer)
  - [Cloud application administrator](../roles/permissions-reference.md#cloud-application-administrator)

> - The tenant-id of the of the Azure Active Directory associated with your Azure Account

Reviewer 2: ```suggestion
- The tenant-id or domain of the of the Azure Active Directory associated with your Azure Account
```

## Create the app using the .NET CLI

```dotnetcli
mkdir <new-project-folder>
cd <new-project-folder>
dotnet new blazorserver --auth SingleOrg --calls-graph
```

## Install the Microsoft Identity App Sync .NET Tool

```dotnetcli
dotnet tool install --global msidentity-app-sync
```

This tool will automate the following tasks for you:

- Register your application in Azure Active Directory
  - Create a secret for your registered application
  - Register redirect URIs based on your launchsettings.json
- Initialize the use of user secrets in your project
- Store your application secret in user secrets storage
- Update your appsettings.json with the client-id, tenant-id, etc...

> > .Net Tools extend the capabilities of the dotnet cli command. To learn more about .Net Tools visit: [https://learn.microsoft.com/en-us/dotnet/core/tools/global-tools](https://learn.microsoft.com/en-us/dotnet/core/tools/global-tools)

Reviewer 2: ```suggestion
> .Net Tools extend the capabilities of the dotnet cli command. To learn more about .Net Tools visit: [https://learn.microsoft.com/dotnet/core/tools/global-tools](https://learn.microsoft.com/dotnet/core/tools/global-tools)
```

> > For more information on user secrets storage please visit: [https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-7.0&tabs=windows](https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-7.0&tabs=windows)

Reviewer 2: ```suggestion
> For more information on user secrets storage please visit: [https://learn.microsoft.com/aspnet/core/security/app-secrets?view=aspnetcore-7.0&tabs=windows](https://learn.microsoft.com/aspnet/core/security/app-secrets?view=aspnetcore-7.0&tabs=windows)
```

## Use the Microsoft Identity App Sync Tool

> Run the following command to register your app in your tenant and update the .Net configuration of your application. Provide the username/upn belonging to your Azure Account and the tenant id of the Azure Active Directory associated with your Azure Account.

Reviewer 2: ```suggestion
Run the following command to register your app in your tenant and update the .Net configuration of your application. Provide the username/upn belonging to your Azure Account (for instance username@domain.com) and the tenant id or domain name of the Azure Active Directory associated with your Azure Account. If you use an account that is signed-in in either Visual Studio, Azure CLI, or Azure PowerShell, you'll benefit from Single Sign On (SSO).
```

Author: From what I could see in my testing.  The browser would pop if you were not already signed in.  It didn't seem like being signed into something else was a pre-req.

Reviewer 2: I've updated the suggestion. Thanks @shoatman

```dotnetcli
msidentity-app-sync --username <username/upn> --tenant-id <tenantID>
```

> 

Reviewer 2: ```suggestion
> [!Note]
> - you don't need to provide the username if you are signed-in with only one account in the developer tools.
> - you don't need to provide the tenant-id if the tenant in which you want to create the application is your home tenant.
```

## Optional - Create a development SSL certificate

In order to avoid SSL errors / warning when browsing the running application you can use the following on macOS and Windows to generate a self-signed SSL certificate for use by .Net core.

```dotnetcli
dotnet dev-certs https --trust
```

## Run the app

In your terminal, run the following command:

```dotnetcli
dotnet run
```

> Browse to the running web application using the URL outputted by the command line.

Reviewer 2: outputted? or output? (base: put, past: put, past participle: put)?

Author: This is one of the marvels of English... it can either be output or outputted.... according to my web searches.... outputted reads more correctly to me.

Run the following command to register your app in your tenant and update the .Net configuration of your application. Provide the username/upn belonging to your Azure Account and the tenant id of the Azure Active Directory associated with your Azure Account.

```dotnetcli
msidentity-app-sync --username <username/upn> --tenant-id <tenantID>
```
> [!Note]
> - you don't need to provide the username if you are signed-in with only one account in the developer tools.
> - you don't need to provide the tenant-id if the tenant in which you want to create the application is your home tenant.
## Optional - Create a development SSL certificate

In order to avoid SSL errors / warning when browsing the running application you can use the following on macOS and Windows to generate a self-signed SSL certificate for use by .Net core.

```dotnetcli
dotnet dev-certs https --trust
```

## Run the app

In your terminal, run the following command:

```dotnetcli
dotnet run
```

Browse to the running web application using the URL outputted by the command line.

## Next steps

Learn about calling building web apps that sign in users in our multi-part scenario series:

> > [!div class="nextstepaction"] > [Scenario: Web app that signs in users](scenario-web-app-sign-user-overview.md)

Reviewer 2: You probably need to have this on 2 lines?

  - [Application administrator](../roles/permissions-reference.md#application-administrator)
  - [Application developer](../roles/permissions-reference.md#application-developer)
  - [Cloud application administrator](../roles/permissions-reference.md#cloud-application-administrator)

> - The tenant-id or domain of the of the Azure Active Directory associated with your Azure Account

Reviewer: ```suggestion
- The tenant-id or domain of the Azure Active Directory associated with your Azure Account
```

## Create the app using the .NET CLI

```dotnetcli
mkdir <new-project-folder>
cd <new-project-folder>
dotnet new blazorserver --auth SingleOrg --calls-graph
```

## Install the Microsoft Identity App Sync .NET Tool

```dotnetcli
dotnet tool install --global msidentity-app-sync
```

This tool will automate the following tasks for you:

- Register your application in Azure Active Directory
  - Create a secret for your registered application
  - Register redirect URIs based on your launchsettings.json
- Initialize the use of user secrets in your project
- Store your application secret in user secrets storage

> - Update your appsettings.json with the client-id, tenant-id, etc...

Reviewer: ```suggestion
- Update your appsettings.json with the client-id, tenant-id, and others.
```

> > .Net Tools extend the capabilities of the dotnet cli command. To learn more about .Net Tools visit: [.NET Tools](/dotnet/core/tools/global-tools)

Reviewer: ```suggestion
.NET Tools extend the capabilities of the dotnet CLI command. To learn more about .NET Tools, see [.NET Tools](/dotnet/core/tools/global-tools).
```

> > For more information on user secrets storage please visit: [safe storage of app secrets during development](/aspnet/core/security/app-secrets)

Reviewer: ```suggestion
For more information on user secrets storage, see [safe storage of app secrets during development](/aspnet/core/security/app-secrets).
```

## Use the Microsoft Identity App Sync Tool

> Run the following command to register your app in your tenant and update the .Net configuration of your application. Provide the username/upn belonging to your Azure Account (for instance username@domain.com) and the tenant id or domain name of the Azure Active Directory associated with your Azure Account. If you use an account that is signed-in in either Visual Studio, Azure CLI, or Azure PowerShell, you'll benefit from Single Sign On (SSO).

Reviewer: ```suggestion
Run the following command to register your app in your tenant and update the .NET configuration of your application. Provide the username/upn belonging to your Azure Account (for instance, username@domain.com) and the tenant ID or domain name of the Azure Active Directory associated with your Azure Account. If you use an account that is signed in in either Visual Studio, Azure CLI, or Azure PowerShell, you'll benefit from Single Sign On (SSO).
```

```dotnetcli
msidentity-app-sync --username <username/upn> --tenant-id <tenantID>
```

> [!Note]

> >
> > - you don't need to provide the username if you are signed-in with only one account in the developer tools.
> > - you don't need to provide the tenant-id if the tenant in which you want to create the application is your home tenant.

Reviewer: ```suggestion
> - You don't need to provide the username if you are signed in with only one account in the developer tools.
> - You don't need to provide the tenant-id if the tenant in which you want to create the application is your home tenant.
```

## Optional - Create a development SSL certificate

> In order to avoid SSL errors / warning when browsing the running application you can use the following on macOS and Windows to generate a self-signed SSL certificate for use by .Net core.

Reviewer: ```suggestion
In order to avoid SSL errors/warnings when browsing the running application, you can use the following on macOS and Windows to generate a self-signed SSL certificate for use by .NET Core.
```

```dotnetcli
dotnet dev-certs https --trust
```

## Run the app

In your terminal, run the following command:

```dotnetcli
dotnet run
```

Browse to the running web application using the URL outputted by the command line.

## Next steps

Learn about calling building web apps that sign in users in our multi-part scenario series:

> [!div class="nextstepaction"]

> 

Reviewer: ```suggestion
```
